File: frontend/src/types/task.ts

// export interface Task {
//   id: string;
//   project_id:string;
//   description: string;
//   title: string;
//   priority: string;
//   status: string;
//   assignee_id: string;
//   due_date: string;
//   created_at: string;
//   embedding_status: boolean
//   // Thêm các trường khác nếu Backend có trả về
// }
// export interface TaskUpdate {
//   title?: string;
//   description?: string;
//   priority?: string;
//   status?: string;
//   due_date?: string ;
//   // Thêm các trường khác nếu Backend có trả về
// }

export enum TaskPriority {
  LOW = "Low",
  MEDIUM = "Medium",
  HIGH = "High",
  URGENT = "Urgent", // Tùy chỉnh theo Enum Backend của bạn
}

export interface Task {
  id: string;
  project_id: string;
  title: string;
  description: string | null;
  priority: TaskPriority | string;
  status: string;
  assignee_id: string | null;
  due_date: string | null;
  created_at: string;
  embedding_status: boolean; // Quan trọng để UI biết AI đã "học" task này chưa
}

export interface TaskCreate {
  project_id: string;
  title: string;
  description?: string;
  priority?: TaskPriority | string;
  status?: string;
  assignee_id?: string;
  due_date?: string;
}

export interface TaskUpdate {
  title?: string;
  description?: string;
  priority?: TaskPriority | string;
  status?: string;
  assignee_id?: string;
  due_date?: string;
}


// --- AI Generate Types ---
export interface TaskAIGenerateRequest {
  project_id: string;
  prompt: string; // Ví dụ: "Lên kế hoạch coding module Auth trong 3 ngày"
}

// --- Semantic Search Types ---
export interface SemanticSearchRequest {
  query: string;
  project_id?: string;
}

// --- RAG Chat Types ---
export interface ChatRequest {
  query: string;
  project_id?: string | null;
}

export interface ChatResponse {
  answer: string;
}

File: backend/app/schemas/task.py
from uuid import UUID
from typing import Optional
from datetime import datetime
from pydantic import BaseModel
from app.models.task import TaskPriority
from uuid import UUID


class TaskCreate(BaseModel):
    project_id: UUID
    title: str
    description: Optional[str] = None
    priority: Optional[TaskPriority] = TaskPriority.MEDIUM
    status: Optional[str] = "To Do"
    assignee_id: Optional[UUID] = None
    due_date: Optional[datetime] = None

class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    priority: Optional[TaskPriority] = None
    status: Optional[str] = None
    assignee_id: Optional[UUID] = None
    due_date: Optional[datetime] = None

class TaskRead(BaseModel):
    id: UUID
    project_id: UUID
    title: str
    description: Optional[str]
    priority: TaskPriority
    status: str
    assignee_id: Optional[UUID]
    due_date: Optional[datetime]
    created_at: datetime
    embedding_status: bool

    class Config:
        from_attributes = True

class TaskAIGenerateRequest(BaseModel):
    project_id: UUID
    prompt: str = "Lên kế hoạch tổ chức workshop AI trong 1 tuần"